pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -119,107 +117,139 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
+	// Add conditions if there are none
+	if instance.Status.Conditions == nil {
+		initConditions(&instance.Status.Conditions)
+
+		err = r.client.Status().Update(context.TODO(), instance)
+		if err != nil {
+			reqLogger.Error(err, "Failed to add conditions to status")
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Create happy conditions
+	initHappyConditions(&r.conditions)
+
 	for _, f := range []func(*hcov1alpha1.HyperConverged, logr.Logger, reconcile.Request) error{
 		r.ensureKubeVirtConfig,
 		r.ensureKubeVirt,
 		r.ensureCDI,
 		r.ensureNetworkAddons,
-		r.ensureKubeVirtCommonTemplatebundle,
+		r.ensureKubeVirtCommonTemplateBundle,
 		r.ensureKubeVirtNodeLabellerBundle,
 		r.ensureKubeVirtTemplateValidator,
 	} {
 		err = f(instance, reqLogger, request)
 		if err != nil {
+			initReconcileFailureConditions(&instance.Status.Conditions)
+			// don't want to overwrite the actual reconcile failure
+			uErr := r.client.Status().Update(context.TODO(), instance)
+			if uErr != nil {
+				reqLogger.Error(uErr, "Failed to update conditions")
+			}
```

Comment:
The HCO will be stuck reconciling the first failure it sees.   Can we avoid that?

Reply:
Admittedly, if we are failing to make it through the `Reconcile()` function something is seriously wrong. That was why I added a new condition type for us `ReconcileComplete` to make it possible to see cases like this where we aren't even making it through the whole Reconcile loop.

Reply:
I don't disagree with you on that point, but do we have to block?  If we do, it can have adverse affects, like not recreating a deleted component cr.  Maybe instead of `ReconcileComplete` there can be `ComponentErrored`, `ComponentMissing`, or `ComponentUknown`?

Reply:
I don't think it's obvious but this isn't really a change in behavior. The only addition is to set the `ReconcileCondition` type to false if we encounter an error.

Reply:
Let's revisit in another PR